include/mxnet/gpu_swap_history.h

```diff
@@ -0,0 +1,56 @@
+#ifndef GPU_SWAP_HISTORY_H_
+#define GPU_SWAP_HISTORY_H_
+
+#include <chrono>
+#include <vector>
+#include "storage.h"
+#include <thread>
+#include <mutex>
+
+#if MXNET_USE_CUDA
+#include <cuda_runtime.h>
+#endif
+
+
+using namespace std::chrono;
+
+namespace mxnet {
+
+class MemHistory {
+public:
+
+  enum record_t {GET_ADDR, SET_ADDR, DEL_ADDR};
+  struct MemRecord {
+    handle_id_t handle_id;
+    record_t operation_id;
+    timestamp_t time;
+    size_t size;
+  };
+
+  std::vector<std::vector<MemRecord> > 
+      history = std::vector<std::vector<MemRecord> >(8);
+  size_t record_idx;
+
+  ~MemHistory();
+  static MemHistory* Get();
+  static std::shared_ptr<MemHistory> _GetSharedRef();
+  bool IterationStarted() {return iteration_started_;}
+  bool DoRecord() {return do_record_;}
```

Comment:
I know I named this function and variable, but the function name is very confusing. "IsRecording" may be a better name.